app/src/main/java/com/hedvig/app/feature/profile/ui/payment/PaymentHistoryAdapter.kt

```diff
@@ -10,21 +10,27 @@ import kotlinx.android.synthetic.main.payment_history_item.view.*
 import kotlinx.android.synthetic.main.payout_history_header.view.*
 import kotlinx.android.synthetic.main.payout_history_title.view.*
 
-class PaymentHistoryAdapter(
-    private val items: List<ChargeWrapper>
-) : RecyclerView.Adapter<PaymentHistoryAdapter.ViewHolder>() {
+class PaymentHistoryAdapter : RecyclerView.Adapter<PaymentHistoryAdapter.ViewHolder>() {
+
+    var items: List<ChargeWrapper> = listOf()
+
     override fun getItemCount() = items.size
 
-    override fun getItemViewType(position: Int) = when (items[position]) {
-        is ChargeWrapper.Header -> HEADER
-        is ChargeWrapper.Item -> ITEM
-        is ChargeWrapper.Title -> TITLE
+    override fun getItemViewType(position: Int) = when (position) {
```

Comment:
I'm not sure that this is correct 🤔 

The previous code was arguably more correct in this case, this might accidentally be correct right now but not later 😉

Reply:
The most recent best practice is to use the Layout ID of the View type for ID, then we do not have to maintain constants in the companion object 🙂